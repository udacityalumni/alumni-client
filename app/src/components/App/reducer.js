import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  user: null,
  isLoading: false,
  error: null,
  navLinks: [
    {
      url: '/careers',
      text: 'Careers',
    },
    {
      url: '/mentorship',
      text: 'Mentorship',
    },
    {
      url: '/meetups',
      text: 'Meetups',
    },
  ],
  navIsActive: false,
  isMobile: false,
  searchTerm: null,
  authToken: null,
};

const appReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.SET_AUTH_TOKEN:
        return update(state, {
          authToken: {
            $set: action.token,
          },
        });
      case types.APP_SET_MOBILE:
        return update(state, {
          isMobile: {
            $set: action.isMobile,
          },
        });
      case types.APP_ON_TOGGLE_NAV:
        return update(state, {
          navIsActive: {
            $set: !state.navIsActive,
          },
        });
      case types.SET_SEARCH_TERM:
        return update(state, {
          searchTerm: {
            $set: action.term,
          },
        });
      case types.CLEAR_SEARCH_TERM:
        return update(state, {
          searchTerm: {
            $set: null,
          },
        });
      case types.SET_AUTH_USER:
        return update(state, {
          user: {
            $set: action.user,
          },
        });
      default:
        return state;
    }
  };


export default appReducer;
