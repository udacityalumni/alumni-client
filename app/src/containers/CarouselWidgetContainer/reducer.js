import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  currentlyEditing: null,
  images: null,
};

const carouselWidgetReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.CAROUSEL_SET_IMAGES:
        return update(state, {
          images: {
            $set: action.images,
          },
        });
      case types.CAROUSEL_SET_EDITING:
        return update(state, {
          currentlyEditing: {
            $set: action.index,
          },
        });
      case types.CAROUSEL_CANCEL_EDITING:
        return update(state, {
          currentlyEditing: {
            $set: null,
          },
        });
      case types.CAROUSEL_ADD_IMAGE:
        return update(state, {
          images: {
            $set: [...state.images, action.image],
          },
        });
      case types.CAROUSEL_REMOVE_IMAGE:
        return update(state, {
          images: {
            $set: [
              ...state.images.slice(0, action.index),
              ...state.images.slice(action.index + 1),
            ],
          },
        });
      case types.CAROUSEL_EDIT_IMAGE:
        return update(state, {
          images: {
            $set: [
              ...state.images.slice(0, action.index),
              action.image,
              ...state.images.slice(action.index + 1),
            ],
          },
          currentlyEditing: {
            $set: null,
          },
        });
      default:
        return state;
    }
  };

export default carouselWidgetReducer;
