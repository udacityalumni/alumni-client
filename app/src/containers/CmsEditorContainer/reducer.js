import * as types from './constants';
import update from 'react-addons-update';
import { editorStateFromRaw } from 'megadraft';

export const initialState = {
  article: {
    id: null,
    action: null,
  },
  editorState: editorStateFromRaw(null),
  editorTitle: null,
  preview: {
    isPreviewing: false,
    content: null,
    title: null,
  },
  isValid: false,
  isLoading: false,
  error: null,
  message: null,
  modal: {
    isShowing: false,
    status: 0,
    spotlighted: false,
    selectedTags: [],
  },
};

const previewReducer = (state = initialState.preview, action) => {
  switch (action.type) {
    case types.CMS_CLOSE_PREVIEW:
      return update(state, {
        isPreviewing: {
          $set: false,
        },
        markdown: {
          $set: null,
        },
        title: {
          $set: null,
        },
      });
    case types.CMS_SET_PREVIEW_STATE:
      return update(state, {
        isPreviewing: {
          $set: true,
        },
        content: {
          $set: action.markdown,
        },
        title: {
          $set: action.title,
        },
      });
    default: return state;
  }
};

const statusEnum = (status) => {
  switch (status) {
    case 'draft':
      return 0;
    case 'published':
      return 1;
    case 'archived':
      return 2;
    default: return 0;
  }
};

const modalReducer = (state = initialState.modal, action) => {
  switch (action.type) {
    case types.CMS_SET_STATE_FROM_ARTICLE:
      return update(state, {
        spotlighted: {
          $set: action.article.spotlighted,
        },
        status: {
          $set: statusEnum(action.article.status),
        },
        selectedTags: {
          $set: action.article.tags.map((tag) =>
            ({
              value: tag.tag,
              label: tag.tag,
            }),
          ),
        },
      });
    case types.CMS_TOGGLE_SPOTLIGHT:
      return update(state, {
        spotlighted: {
          $set: !state.spotlighted,
        },
      });
    case types.CMS_OPEN_MODAL:
      return update(state, {
        isShowing: {
          $set: true,
        },
      });
    case types.CMS_CLOSE_MODAL:
      return update(state, {
        isShowing: {
          $set: false,
        },
      });
    case types.CMS_SET_STATUS:
      return update(state, {
        status: {
          $set: action.status,
        },
      });
    case types.CMS_SET_SELECTED_TAGS:
      return update(state, {
        selectedTags: {
          $set: action.tags,
        },
      });
    default: return state;
  }
};

const cmsEditorReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.SUBMIT_ARTICLE_INITIATION:
        return update(state, {
          isLoading: {
            $set: true,
          },
        });
      case types.SUBMIT_ARTICLE_FAILURE:
        return update(state, {
          isLoading: {
            $set: false,
          },
          error: {
            $set: action.error,
          },
        });
      case types.SUBMIT_ARTICLE_SUCCESS:
        return update(state, {
          isLoading: {
            $set: false,
          },
          message: {
            $set: action.message,
          },
        });
      case types.CLEAR_CMS_ERROR:
        return update(state, {
          error: {
            $set: null,
          },
        });
      case types.CLEAR_CMS_MESSAGE:
        return update(state, {
          message: {
            $set: null,
          },
        });
      case types.CMS_OPEN_MODAL:
        return update(state, {
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      case types.CMS_CLOSE_MODAL:
        return update(state, {
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      case types.CMS_SET_STATUS:
        return update(state, {
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      case types.CMS_SET_SELECTED_TAGS:
        return update(state, {
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      case types.CMS_TOGGLE_SPOTLIGHT:
        return update(state, {
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      case types.CMS_SET_EDITOR_TITLE:
        return update(state, {
          editorTitle: {
            $set: action.title,
          },
          isValid: {
            $set: action.title !== '',
          },
        });
      case types.CMS_SET_EDITOR_STATE:
        return update(state, {
          editorState: {
            $set: action.state,
          },
        });
      case types.CMS_CLOSE_PREVIEW:
        return update(state, {
          preview: {
            $set: previewReducer(state.preview, action),
          },
        });
      case types.CMS_SET_PREVIEW_STATE:
        return update(state, {
          preview: {
            $set: previewReducer(state.preview, action),
          },
        });
      case types.CMS_SET_ARTICLE_ID:
        return update(state, {
          article: {
            id: {
              $set: parseInt(action.id, 10),
            },
            action: {
              $set: action.action,
            },
          },
        });
      case types.CMS_SET_STATE_FROM_ARTICLE:
        const rawContent = JSON.parse(action.article.json);
        return update(state, {
          isValid: {
            $set: true,
          },
          editorState: {
            $set: editorStateFromRaw(rawContent),
          },
          editorTitle: {
            $set: action.article.title,
          },
          modal: {
            $set: modalReducer(state.modal, action),
          },
        });
      default:
        return state;
    }
  };

export default cmsEditorReducer;
