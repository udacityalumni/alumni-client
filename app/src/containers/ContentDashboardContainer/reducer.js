import * as types from './constants';
import update from 'react-addons-update';

export const initialState = {
  articles: null,
  error: null,
  message: null,
  isLoading: false,
  isShowingModal: false,
  selectedArticleId: null,
};

const contentDashboardReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.DASHBOARD_DELETE_ARTICLE_INITIATION:
        return update(state, {
          isLoading: {
            $set: true,
          },
        });
      case types.DASHBOARD_DELETE_ARTICLE_SUCCESS:
        return update(state, {
          isLoading: {
            $set: false,
          },
          message: {
            $set: action.message,
          },
          selectedArticleId: {
            $set: null,
          },
        });
      case types.DASHBOARD_DELETE_ARTICLE_FAILURE:
        return update(state, {
          isLoading: {
            $set: false,
          },
          error: {
            $set: action.error.message,
          },
          selectedArticleId: {
            $set: null,
          },
        });
      case types.DASHBOARD_TOGGLE_MODAL:
        return update(state, {
          isShowingModal: {
            $set: true,
          },
          selectedArticleId: {
            $set: action.id,
          },
        });
      case types.DASHBOARD_MODAL_CONFIRMATION:
        return update(state, {
          isShowingModal: {
            $set: false,
          },
          selectedArticleId: {
            $set: null,
          },
        });
      case types.DASHBOARD_MODAL_CANCEL:
        return update(state, {
          isShowingModal: {
            $set: false,
          },
          selectedArticleId: {
            $set: null,
          },
        });
      case types.DASHBOARD_ARTICLES_INITIATION:
        return update(state, {
          isLoading: {
            $set: true,
          },
        });
      case types.DASHBOARD_ARTICLES_SUCCESS:
        return update(state, {
          isLoading: {
            $set: false,
          },
          articles: {
            $set: action.articles,
          },
        });
      case types.DASHBOARD_ARTICLES_FAILURE:
        return update(state, {
          isLoading: {
            $set: false,
          },
          error: {
            $set: action.error,
          },
        });
      case types.CLEAR_DASHBOARD_ERROR:
        return update(state, {
          error: {
            $set: null,
          },
        });
      case types.CLEAR_DASHBOARD_MESSAGE:
        return update(state, {
          message: {
            $set: null,
          },
        });
      default:
        return state;
    }
  };

export default contentDashboardReducer;
