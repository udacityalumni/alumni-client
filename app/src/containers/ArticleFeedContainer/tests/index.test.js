import ArticleFeed from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialState as articleFeedContainer } from '../reducer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<ArticleFeed />', () => {
  it('should render with default props', () => {
    const store = mockStore({ articleFeedContainer });
    const wrapper = shallow(
      <ArticleFeed store={store} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
