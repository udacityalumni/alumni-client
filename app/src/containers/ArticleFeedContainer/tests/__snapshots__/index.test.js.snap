exports[`<ArticleFeed /> should render with default props 1`] = `
<Apollo(ArticleFeed)
  actions={
    Object {
      "articleFeedIncrementCurrent": [Function anonymous]
    }
  }
  current={3}
  store={
    Object {
      "clearActions": [Function clearActions],
      "dispatch": [Function anonymous],
      "getActions": [Function getActions],
      "getState": [Function getState],
      "replaceReducer": [Function replaceReducer],
      "subscribe": [Function subscribe]
    }
  } />
`;
